src/aap_eda/wsapi/consumers.py

```diff
@@ -513,3 +512,43 @@ def get_file_contents_from_credentials(
                     contents = substitute_variables(value, inputs)
                     file_templates[template] = contents
         return file_templates
+
+    @database_sync_to_async
+    def get_env_vars_from_credentials(
+        self, activation: models.Activation
+    ) -> tp.Optional[str]:
+        env_vars = {}
+        vault_inputs = {}
+        if activation.eda_system_vault_credential:
+            vault_inputs = activation.eda_system_vault_credential.inputs
+            vault_inputs = yaml.safe_load(vault_inputs.get_secret_value())
+
+        for eda_credential in activation.eda_credentials.all():
+            schema_inputs = eda_credential.credential_type.inputs
+            secret_fields = get_secret_fields(schema_inputs)
+            injectors = eda_credential.credential_type.injectors
+            user_inputs = yaml.safe_load(
+                eda_credential.inputs.get_secret_value()
+            )
+            if "env" in injectors:
+                for key, value in user_inputs.items():
+                    if key in secret_fields:
+                        user_inputs[key] = encrypt_string(
+                            password=vault_inputs["vault_password"],
```

Comment:
if `secret_fields` exist then the `eda_system_vault_credential` must exist via activation creation process

Reply:
@bzwei Since we know that RHAAP credential type has env we have to set this flag https://github.com/ansible/eda-server/blob/1c948a67f45c1d1229b9abe6724b81dd1c5c2307/src/aap_eda/api/serializers/activation.py#L148 otherwise if someone attaches RHAPP to Activation it will fail